src/app/components/admin-app/app-admin-subscription-search/app-admin-subscription-search.component.ts

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2021 Carnegie Mellon University. All Rights Reserved. 
+ Released under a MIT (SEI)-style license. See LICENSE.md in the project root for license information.
+*/
+
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatSort, MatSortable } from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material/table';
+import { WebhookService, WebhookSubscription } from '../../../generated/player-api';
+import { DialogService } from '../../../services/dialog/dialog.service';
+
+@Component({
+  selector: 'app-admin-subscription-search',
+  templateUrl: './app-admin-subscription-search.component.html',
+  styleUrls: ['./app-admin-subscription-search.component.scss']
+})
+export class AppAdminSubscriptionSearchComponent implements OnInit {
+  @ViewChild(MatSort, { static: true }) sort: MatSort;
+
+  public dataSource: MatTableDataSource<WebhookSubscription>;
+  public displayedColumns: string[] = ['name', 'lastError', 'eventTypes'];
+  public filterString: string;
+  public editing: boolean = false;
+  public filterStr: string;
+
+  constructor(
+    private webhookService: WebhookService,
+    private dialogService: DialogService,
+  ) {}
+  ngOnInit(): void {
+    // Initialize table
+    this.dataSource = new MatTableDataSource<WebhookSubscription>(new Array<WebhookSubscription>());
+    this.sort.sort(<MatSortable>{ id: 'name', start: 'asc' });
+    this.dataSource.sort = this.sort;
+    this.filterString = '';
+
+    this.refreshSubs();
+  }
+
+  addNewSubscription() {
+    this.dialogService.editSubscription().subscribe(resp => {
+      console.log(resp);
+      this.refreshSubs();
+    })
+  }
+
+  refreshSubs() {
+    this.webhookService.getAll().subscribe((subs) => {
```

Comment:
".subscribe" should be replaced by ".pipe(takeUntil(this.unsubscribe$)).subscribe" and an ngOnDestroy() function should be added.  An example can be seen in logged-in-user.service.ts line 42 and following or app.component.ts lines 30 and 153.